clippy_lints/src/loops/mod.rs

```diff
@@ -397,6 +397,10 @@ declare_clippy_lint! {
     /// ### Why is this bad?
     /// One might think that modifying the mutable variable changes the loop bounds
     ///
+    /// ### Known problems
+    /// False positive when mutation is not immediately followed by a break statement
+    /// False positive on nested expressions ([#6072](https://github.com/rust-lang/rust-clippy/issues/6072))
```

Comment:
```suggestion
    /// False positive when mutation is followed by a `break`, but the `break` is not immediately after the mutation:
    ///
    /// ```
    /// x += 1; // x is range bound
    /// some_other_expr();
    /// break; // leaves the loop so mutating is not an issue
    /// ```
    ///
    /// False positive on nested loops ([#6072](https://github.com/rust-lang/rust-clippy/issues/6072))
```